docs/DeveloperGuide.adoc

```diff
@@ -319,6 +319,145 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
+// tag::recommend[]
+=== Recommending a workout
+==== Current implementation
+The user input is parsed into invoking a constructor based on the prefixes in the user input in `_RecommendCommandParser_`.There are only three acceptable prefixes. (i.e. `difficulty`, `calories`, `duration`) The instance created is then passed to `_RecommendCommand_` which uses `getFilteredWorkoutList()` and `getFilteredInternalList()` methods of `Model` to generate a random workout to recommend to the user.
+
+The sequence diagram of this entire process can be seen below:
+
+[[fig-RecommendCommandSD]]
+.Interactions inside `Logic` component for the *`recommend difficulty/beginner`* command.
+image::RecommendCommandSD.PNG[width="900"]
+
+==== Design Considerations
+
+==== Aspect: How recommend executes
+
+* **Alternative 1 (current choice):** Filters the workout list based on user's input and randomly generates a workout from that filtered list.
+** Pros: Easy to implement.
+** Cons: May not reflect what the user actually wants.
+* **Alternative 2:** Use machine learning algorithms.
+** Pros: Able to recommend a workout that is more accurate to what the user wants.
+** Cons: Complicated to implement. Requires a huge storage to store and record the substantial amount of data that comes from the interactions between the user and the application.
+// end::recommend[]
+
+=== User Profile
+// tag::profile
+==== Current implementation
+The profile window is first being loaded by loading the `_profileWindow.fxml file_`. Thereafter, once the user request to open the profile window, it will then process it.
+
+There are 3 ways to open a profile window.
+
+* User types `"profile"` in command box
+* User clicks on `profile` on the menu bar
+* User presses the `F2` key
+
+===== User types "profile" in command box
+
+After executing the command `profile`, it will create an instance which uses `showProfileRequestEvent` method of `Events`. Then, the `UI` will `handleShowProfileRequestEvent` and the profile window will be opened.
+
+[[fig-ProfileCommand]]
+.Interactions inside `Logic` component for the *`profile`* command
+image::ProfileCommand.PNG[width="800"]
```

Comment:
I am confused about some of the events in this sequence diagram.  1) If showProfileRequestEvent() is shown to create an "Events" object, then shouldn't there be a constructor. 2) Some of the processes seem to be working in "parallel", e.g. CommandResult is invoked before control returns from "Events". This may not be true as the code that you are working with are working sequentially.  Do give a look into it. Please ignore, if I understood wrongly.